api/ephemeral-access/v1alpha1/accessrequest_types.go

```diff
@@ -77,16 +77,23 @@ type TargetApplication struct {
 // TargetRole defines the role that is requested
 type TargetRole struct {
 	// TemplateName defines the role template the user will be assigned
-	// +kubebuilder:validation:MaxLength=512
 	// +kubebuilder:validation:Required
-	TemplateName string `json:"templateName"`
+	Template TargetRoleTemplate `json:"template"`
```

Comment:
maybe the spec field should be `templateRef`

Reply:
done